connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -700,9 +714,28 @@ private class HandleRebalance implements ConsumerRebalanceListener {
         @Override
         public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
             log.debug("{} Partitions assigned {}", WorkerSinkTask.this, partitions);
-
+            Set<String> deletedTopics = new HashSet<>();
             for (TopicPartition tp : partitions) {
-                long pos = consumer.position(tp);
+                if (deletedTopics.contains(tp.topic())) {
+                    log.debug("Not Committing offsets for topic-partition {} since the topic {} has been deleted", tp, tp.topic());
+                    continue;
+                }
+                long pos;
+                try {
+                    pos = consumer.position(tp);
+                } catch (TimeoutException e) {
```

Comment:
This seems kinda weird to me too - why would the consumer time out on trying to fetch the position for a topic partition that doesn't exist anymore? Is this expected / documented or just a poorly handled case in the consumer client?

Reply:
It works under normal scenarios. I tested by deleting 10 topics in one go and still didn't notice this error happening. In this case, when a large number of topics gets deleted is when we see this error. Note that I don't know what is the defintion of large but going by the update of the reporter of the ticket, they experience it with a few dozens of topics deleted.

Reply:
I had done some analysis and a plausible explanation this comment onwards: https://issues.apache.org/jira/browse/KAFKA-14750?focusedCommentId=17710077&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17710077